client/components/sites-dropdown/index.jsx

```diff
@@ -38,20 +40,23 @@ export default class SitesDropdown extends PureComponent {
 		super( props );
 
 		this.selectSite = this.selectSite.bind( this );
-		this.toggleOpen = this.toggleOpen.bind( this );
+		this.toggleDropdown = this.toggleDropdown.bind( this );
 		this.onClose = this.onClose.bind( this );
 
-		const selectedSite = props.selectedSiteId
-			? sites.getSite( props.selectedSiteId )
-			: sites.getPrimary();
+		const selectedSite = props.primarySite || sites.getPrimary();
 
 		this.state = {
 			selectedSiteSlug: selectedSite && selectedSite.slug
 		};
 	}
 
-	getSelectedSite() {
-		return sites.getSite( this.state.selectedSiteSlug );
+	componentWillReceiveProps( nextProps ) {
+		const { primarySite } = nextProps;
+		if ( primarySite && primarySite.slug !== this.state.selectedSiteSlug ) {
+			this.setState( {
+				selectedSiteSlug: primarySite.slug
+			} );
```

Comment:
When we're ready, I would instead move this logic to `getSelectedSiteSlug`  or `getSelectedSiteId` in `state/sites/selectors`. Unless there's a special need, this can also be passed through as a connected prop vs setting this as component state.